src/features/game/actions/loadFrogs.ts

```diff
@@ -0,0 +1,128 @@
+import { metamask } from "lib/blockchain/metamask";
+import { CONFIG } from "lib/config";
+import frog_unrevealed from "../../../assets/nfts/frogs/frog_unrevealed.gif";
+
+const FROG_URL = CONFIG.FROG_URL;
+
+type Request = {
+  frogId: number[];
+};
+
+export type Frog = {
+  pixel_image: string;
+  attributes?: any;
+  [key: string]: any;
+};
+
+export async function loadFrogs() {
+  let frogIds = await metamask.getFrog().getFrogIds();
+
+  // for testing
+  // if (false) {
+  if (await isFrogPreview()) {
+    if (frogIds.length > 6) {
+      frogIds = frogIds.slice(0, 6);
+    }
+
+    const result = frogIds.map(() => ({
+      pixel_image: frog_unrevealed,
+      name: "Unrevealed",
+    }));
+
+    return result;
+  } else {
+    const result = frogIds.map(
+      async (id) => await getFrogMetadata({ frogId: id })
+    );
+    const res: Frog[] = await Promise.all(result);
+
+    let filteredFrogs = res;
+
+    // only run filter if there is more than 1 frog in wallet
+    if (frogIds.length > 1) {
+      filteredFrogs = filterFrogs(res);
+    }
+
+    console.log("filtered frogs:", filteredFrogs);
+    return filteredFrogs;
+  }
+}
+
+export async function getFrogMetadata(request: Request) {
+  const headers = new Headers();
+
+  headers.append("Content-Type", "application/json");
+  headers.append("Accept", "application/json");
+
+  const url = `${FROG_URL}/frogs/${request.frogId}`;
```

Comment:
To follow ERC721 standards, we should fetch this URL directly from the smart contract instead of hard coding the `FROG_URL`.

For instance.

`const url = metamask.getFrog().getTokenUri(id)`

This would return the URL from the on-chain data

Reply:
Hi adam!

The frog URI (metadata URL) is different from the `FROG_URL` (which is the API) I think I would have named it better as FROG_API, I think it would be saver adding it to the .env file rather than adding it to the contract. What do you think? 😅

Reply:
Done! ✌🏻😄